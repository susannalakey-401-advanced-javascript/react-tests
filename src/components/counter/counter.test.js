import React from 'react'
import Counter from './counter'
import { shallow } from 'enzyme'


describe('<Counter />', () => {
  let tree;

  beforeEach(() => {
    tree = shallow(<Counter />)
  })

  it('renders to the page', () => {
    expect(tree).toMatchSnapshot();
  })

  describe('when the down is clicked', () => {

    beforeEach(() => {

      // find the element by targeting it, then simulate the click
      tree.find('.down').simulate('click');

    })

    it('renders to the page', () => {
      expect(tree).toMatchSnapshot();
    })

  })


  describe('when the up is clicked', () => {

    beforeEach(() => {
      // find the element by targeting it, then simulate the click
      tree.find('.up').simulate('click');
    })

    it('renders to the page', () => {
      expect(tree).toMatchSnapshot();
    })

  })

})





// describe('<Thing />', () => {
//   it('renders correctly', () => {
//     const tree = renderer.create(<Thing />).toJSON()
//     expect(tree).toMatchSnapshot()
//   })

//   it('changes state on click', () => {
//     const component = mount(<Thing />)
//     const button = component.find('#myButton')
//     button.simulate('click')
//     expect(component).toHaveState({ stuff: false })
//     expect(component.find('span')).toIncludeText('false')
//   })
// })


// describe('<App />', () => {
//   it('renders correctly', () => {
//     const tree = renderer.create(<App />).toJSON()
//     expect(tree).toMatchSnapshot()
//   })

//   it('is alive at application start', () => {
//     const component = shallow(<App />)
//     expect(component.find('h1')).toExist()
//   })
// })