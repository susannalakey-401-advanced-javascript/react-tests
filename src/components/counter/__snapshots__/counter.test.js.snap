// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders to the page 1`] = `
<section
  className="counter"
>
  <span
    className="down clicker"
    onClick={[Function]}
  >
     - 
  </span>
  <span
    className="count"
  >
    0
  </span>
  <span
    className="up clicker"
    onClick={[Function]}
  >
     + 
  </span>
</section>
`;

exports[`<Counter /> when the down is clicked renders to the page 1`] = `
<section
  className="counter"
>
  <span
    className="down clicker"
    onClick={[Function]}
  >
     - 
  </span>
  <span
    className="count"
  >
    -1
  </span>
  <span
    className="up clicker"
    onClick={[Function]}
  >
     + 
  </span>
</section>
`;

exports[`<Counter /> when the up is clicked renders to the page 1`] = `
<section
  className="counter"
>
  <span
    className="down clicker"
    onClick={[Function]}
  >
     - 
  </span>
  <span
    className="count"
  >
    1
  </span>
  <span
    className="up clicker"
    onClick={[Function]}
  >
     + 
  </span>
</section>
`;
